core/src/main/java/io/aiven/klaw/service/OperationalRequestsService.java

```diff
@@ -0,0 +1,337 @@
+        operationalRequestModel.setApprovingTeamDetails(
+            updateApproverInfo(
+                userList,
+                manageDatabase.getTeamNameFromTeamId(tenantId, userTeamId),
+                approverRoles,
+                operationalRequestModel.getRequestor()));
+      }
+      operationalRequestModelList.add(setRequestorPermissions(operationalRequestModel, userName));
+    }
+    return operationalRequestModelList;
+  }
+
+  private OperationalRequestsResponseModel setRequestorPermissions(
+      OperationalRequestsResponseModel req, String userName) {
+    log.debug(
+        " My request Status {} and userName {} and requestor {}",
+        req.getRequestStatus(),
+        userName,
+        req.getRequestor());
+    if (RequestStatus.CREATED == req.getRequestStatus()
+        && userName != null
+        && userName.equals(req.getRequestor())) {
+      req.setDeletable(true);
+      req.setEditable(true);
+    }
+
+    return req;
+  }
+
+  private String updateApproverInfo(
+      List<UserInfo> userList, String teamName, List<String> approverRoles, String requestor) {
+    StringBuilder approvingInfo = new StringBuilder("Team : " + teamName + ", Users : ");
+
+    for (UserInfo userInfo : userList) {
+      if (approverRoles.contains(userInfo.getRole())
+          && !Objects.equals(requestor, userInfo.getUsername())) {
+        approvingInfo.append(userInfo.getUsername()).append(",");
+      }
+    }
+
+    return approvingInfo.toString();
+  }
+
+  private List<OperationalRequest> getOperationalRequestsPaged(
+      List<OperationalRequest> origActivityList, String pageNo, String currentPage) {
+
+    List<OperationalRequest> newList = new ArrayList<>();
+    Env envSelected;
+
+    if (origActivityList != null && origActivityList.size() > 0) {
+      int totalRecs = origActivityList.size();
+      int recsPerPage = 10;
+      int totalPages = totalRecs / recsPerPage + (totalRecs % recsPerPage > 0 ? 1 : 0);
+
+      pageNo = commonUtilsService.deriveCurrentPage(pageNo, currentPage, totalPages);
+      int requestPageNo = Integer.parseInt(pageNo);
+      int startVar = (requestPageNo - 1) * recsPerPage;
+      int lastVar = (requestPageNo) * (recsPerPage);
+
+      List<String> numList = new ArrayList<>();
+      commonUtilsService.getAllPagesList(pageNo, currentPage, totalPages, numList);
+
+      for (int i = Math.max(0, startVar); i < Math.min(lastVar, totalRecs); i++) {
+        OperationalRequest operationalRequest = origActivityList.get(i);
+        operationalRequest.setAllPageNos(numList);
+        operationalRequest.setTotalNoPages("" + totalPages);
+        operationalRequest.setCurrentPage(pageNo);
+        envSelected = getEnvDetails(operationalRequest.getEnvironment());
+        operationalRequest.setEnvironmentName(envSelected.getName());
+        newList.add(operationalRequest);
+      }
+    }
+
+    return newList;
+  }
+
+  public Env getEnvDetails(String envId) {
+    Optional<Env> envFound =
```

Comment:
Could we replace this with commonUtilsService.getEnvDetails(envId,commonUtilsService.getTenantId(getUserName())); ?

Reply:
Updated. Probably we need to update this in other services..future PRs